### PR TITLE
Switch from Gatsby to being part of new DevHub

### PR DESCRIPTION
**Pull request by MonicaG**
This pull request merges all the changes from the [bcgov/private-cloud-techdocs](https://github.com/bcgov/private-cloud-techdocs) repo. This fork was used while we ran in a parallel state with the content on both the docs.developer.gov.bc.ca site and the new [DevHub site](https://developer.gov.bc.ca).

The following changes were made:
* Removed all Gatsby related code
* Formatted content to display within DevHub nicely
* Added files needed to work with DevHub (catalog-info.yaml, mkdocs.yml, index.md)
* Added a GitHub workflow that will publish the content to the dev site on a PR and to the prod site on a push to main

### CODE REVIEW

File: src/docs/build-deploy-and-maintain-apps/image-artifact-management-with-artifactory.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ sort_order: 5
 
 # Image and artifact management with Artifactory
 Last updated: **January 29, 2024**
+Last updated: **January 29, 2024**
```
Comment (mtspn) [inline src/docs/build-deploy-and-maintain-apps/image-artifact-management-with-artifactory.md:23]:
```suggestion
```
Remove duplicate last updated field

File: src/docs/build-deploy-and-maintain-apps/push-pull-artifacts-artifactory.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Pull artifacts from Artifactory
+title: Pull artifacts from Artifactory
```
Comment (mtspn) [inline src/docs/build-deploy-and-maintain-apps/push-pull-artifacts-artifactory.md:3]:
```suggestion
```
delete duplicate title field

File: src/docs/build-deploy-and-maintain-apps/setup-artifactory-project-repository.md [hunk 3]
```diff
@@ -15,6 +15,7 @@ author: Jonathan Bond
 
 content_owner: Cailey Jones
 
+sort_order: 9
```
Comment (mtspn) [inline src/docs/build-deploy-and-maintain-apps/setup-artifactory-project-repository.md:18]:
```suggestion
```
remove duplicate sort order

File: src/docs/build-deploy-and-maintain-apps/setup-artifactory-service-account.md [hunk 4]
```diff
@@ -15,6 +15,7 @@ author: Jonathan Bond
 
 content_owner: Cailey Jones
 
+sort_order: 8
```
Comment (mtspn) [inline src/docs/build-deploy-and-maintain-apps/setup-artifactory-service-account.md:18]:
```suggestion
```
remove duplicate sort order

**Review by ShellyXueHan — APPROVED** [review]
LGTM!

File: src/docs/reusable-code-and-services/reusable-services-list.md [hunk 5]
```diff
@@ -85,15 +87,15 @@ You can get an unrestricted APIkey for use in government applications by opening
...
-    -  Owasp scan
+  - Installation guide on how to setup pipeline on either your local workspace or from a docker container
+  -  React app build and deploy
+  -  Using builda to build and push an image
+  -  Build and deploy application via helm
+  -  Maven build
+  -  Codeql scan
+  -  Sonar scan
+  -  Trivy scan
+  -  Owasp scan
```
Comment (mtspn) [inline src/docs/reusable-code-and-services/reusable-services-list.md:98]:
```suggestion

  - Installation guide on how to setup pipeline on either your local workspace or from a docker container
  -  React app build and deploy
  -  Using builda to build and push an image
  -  Build and deploy application via helm
  -  Maven build
  -  Codeql scan
  -  Sonar scan
  -  Trivy scan
  -  Owasp scan
```
Is a line break needed here for this list to work?

**Review by mtspn — CHANGES_REQUESTED** [review]
Overall looks really good, just a few suggestions that I noticed.

File: src/docs/reusable-code-and-services/reusable-services-list.md (hunk 5 above)
Comment (MonicaG) [reply src/docs/reusable-code-and-services/reusable-services-list.md:98]: Not in this case where it is a nested list. 

There is a line break between the `CI/CD Pipeline Templates and GitHub Actions templates` intro sentence and the first element `- OpenShift Pipeline (aka Tekton) examples:` which is needed.

But for a list, which has a sublist, the extra space is not needed between the elements. In this case the `- OpenShift Pipeline (aka Tekton) examples:` element and the sub element `- Installation guide on how to setup pipeline on either your local workspace or from a docker container` does NOT need a line break between them.

**Comment by MonicaG** [comment]
> Overall looks really good, just a few suggestions that I noticed.

Thank you @mtspn ! I've removed the duplicate lines as you suggested. Let me know if my above comment on [reusable-services-list.md](https://github.com/bcgov/platform-developer-docs/pull/229/files/393a8ea37bbabb197b9446967083a83d38dbd3da#diff-4cc6e9bc8a7c95b37abfdafa3e7823b83bf8350949e791fb70ecc312880df04b) makes sense or not.

**Review by mtspn — APPROVED** [review]
Thanks Monica, makes sense about the nested lists. Looks good to me.

**Comment by MonicaG** [comment]
FYI: I believe the "Build TechDocs with DevHub TechDocs Publish Action" workflow is failing because it requires access to secrets in this repo. Because this PR is from a fork, GitHub is denying access to those secrets. Once the code is merged, the workflow will have access to the secrets and should run successfully.

**Pull request merged**